python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -312,6 +324,26 @@ def _report_current_usage(
         if self._global_info:
             self._global_info.set_description(resources_status)
 
+    def _update_dataset_metrics(self):
+        stats = {metric: 0 for metric in DataMetric}
+
+        for op in self._topology:
+            metrics = op.metrics
+            resource_usage = op.current_resource_usage()
+
+            stats[DataMetric.BYTES_SPILLED] += metrics.obj_store_mem_spilled
+            stats[DataMetric.BYTES_ALLOCATED] += metrics.obj_store_mem_alloc
+            stats[DataMetric.BYTES_FREED] += metrics.obj_store_mem_freed
+            # Input operators only produce outputs and do not take inputs,
+            # so this metric is not properly calculated in these cases
+            if not isinstance(op, InputDataBuffer):
+                stats[DataMetric.BYTES_CURRENT] += metrics.obj_store_mem_cur
```

Comment:
This should be easy to fix. Just add `self._metrics.on_input_received` in `InputDataBuffer.start`